tools/c7n_azure/c7n_azure/lookup.py

```diff
@@ -0,0 +1,68 @@
+# 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import jmespath
+from jsonschema import validate, exceptions
+
+
+class Lookup(object):
```

Comment:
Feels like this should be in c7n core.  

I think if we have no use cases for it there today it is fine to merge it here and then move it there as we work on secrets support.

Reply:
Yeah I discussed offline yesterday with the team that we'll check it in here and then make the argument to move it to c7n core.